app/Http/Controllers/SupportController.php

```diff
@@ -346,24 +346,48 @@ public function checkIfSupportExist(Request $request)
                 return $this->resProvider->apiJsonResponse(400, trans('message.error.exception'), '','');
             }
 
-             /* First check Support exists or not*/
-             $DelegateSupport = Support::checkIfDelegateSupportExists($topicNum, $nickNames, $delegataedNickNameId);
-             if($DelegateSupport){
+            
+             if(isset($delegataedNickNameId) && $delegataedNickNameId)
+             {
+                 /* First check Support exists or not*/
+                 $DelegateSupport = Support::checkIfDelegateSupportExists($topicNum, $nickNames, $delegataedNickNameId);
+                 if($DelegateSupport)
+                 {
+     
+                     $nickName = Nickname::getNickName($delegataedNickNameId);
+                     $returnData['warning'] =  $nickName->nick_name . " is already delegating support to you, you cannot delegate your support to this user";
+                     $returnData['is_delegator'] = 0;
+                     $returnData['topic_num'] = $topicNum;
+                     $returnData['camp_num'] = $campNum;
+                     $returnData['delegated_nick_name_id'] = $nickName->id;
+                     $returnData['is_confirm'] = 1;
+                     $returnData['disable_submit'] = 1;
+                     $returnData['nick_name_link'] = Nickname::getNickNameLink($delegataedNickNameId, '1', $topicNum, $campNum);
+                     $returnData['remove_camps'] = [];
+     
+                     return $this->resProvider->apiJsonResponse(200, trans('message.success.success'), $returnData, '');
+                     
+                 }
+ 
+                 //check if having direct direct support
+                 $support = Support::checkIfSupportExists($topicNum, [$delegataedNickNameId],[$campNum]); 
+                 if(empty($support))
+                 {
+                     $nickName = Nickname::getNickName($delegataedNickNameId);
```

Comment:
@akuks  these are not same we checking here in case of delegating support does that user have direct support 

and another case w are checking if directly supporting user have already supported this camp. there are two differrent ids passed.

Reply:
@nalwa lines looks like exactly the same. We can move it to a separate function and then `return $returnData`